mount/upgrader.go

```diff
@@ -18,19 +21,20 @@ import (
 type Upgrader struct {
 	underlying  Mount
 	passthrough bool
+	shardKey    shard.Key
 
-	lk        sync.Mutex
-	transient string
-	rootdir   string
+	lk      sync.Mutex
+	refs    uint32
+	rootdir string
 }
 
 var _ Mount = (*Upgrader)(nil)
 
 // Upgrade constructs a new Upgrader for the underlying Mount. If provided, it
 // will reuse the file in path `initial` as the initial transient copy. Whenever
 // a new transient copy has to be created, it will be created under `rootdir`.
-func Upgrade(underlying Mount, rootdir, initial string) (*Upgrader, error) {
-	ret := &Upgrader{underlying: underlying, rootdir: rootdir}
+func Upgrade(underlying Mount, rootdir string, shardKey shard.Key, initial string) (u *Upgrader, finalErr error) {
```

Comment:
This shouldn't depend on package `shard`. Replacing the type of `key` for `string` is better.